src/components/legend/Bubbles.js

```diff
@@ -15,10 +17,23 @@ const digitWidth = 6.8
 export const guideLength = 16
 export const textPadding = 4
 
-const Bubbles = ({ radiusLow, radiusHigh, color, classes }) => {
+const Bubbles = ({
+    radiusLow,
+    radiusHigh,
+    color,
+    minValue,
+    maxValue,
+    classes,
+}) => {
+    const noData = classes.find((c) => c.noData === true)
+    const filteredClasses = classes.filter((c) => !c.noData)
+
+    const binSize = (maxValue - minValue) / (filteredClasses.length || 3)
```

Comment:
General thought about this component. There is a lot of logic that could easily be tested with jest if it was in a separate module. I guess it could also be jest tested using testing library. I'm partly inclined to suggest extracting all the logic to an external function for testing, but I'm not sure. What do you think?